test/jdk/javax/swing/JPopupMenu/bug4234793.java

```diff
@@ -0,0 +1,255 @@
+        public void mouseClicked(MouseEvent ex) {
+        }
+
+        private void maybeShowPopup(MouseEvent e) {
+            if (e.isPopupTrigger()) {
+                popup.show(e.getComponent(), e.getX(), e.getY());
+            }
+        }
+    }
+
+    static class PopupHandler extends AbstractAction{
+        public void actionPerformed(ActionEvent e){
+            if(!popupMenu.isVisible())
+                popupMenu.show((Component)e.getSource(),40,40);
+        }
+    }
+
+    static JPanel createContentPane() {
+        popupMenu = new JPopupMenu();
+        JMenuItem item;
+        for (int i = 0; i < dayData.length; i++) {
+            item = popupMenu.add(new JMenuItem(dayData[i], mnDayData[i]));
+        }
+        popupMenu.addPopupMenuListener(frame);
+
+        JComboBox combo = new JComboBox(numData);
+        combo.addPopupMenuListener(frame);
+        JPanel comboPanel = new JPanel();
+        comboPanel.add(combo);
+
+        //JLabel message = new JLabel("Message area");
+
+        JPanel panel = new JPanel(new BorderLayout());
+
+        panel.add(new JLabel("Right click on the panel to show the PopupMenu"), BorderLayout.NORTH);
+        panel.add(comboPanel, BorderLayout.CENTER);
+        //panel.add(message, BorderLayout.SOUTH);
+
+        return panel;
+    }
+
+    public void setMessage(String msg) {
+        PassFailJFrame.log(msg);
+        //System.out.println(msg);
+
+        //message.setText(msg);
+        //message.repaint();
+    }
+
+
+    static JMenuBar createMenuBar() {
+        JMenuBar menubar = new JMenuBar();
+        JMenuItem menuitem;
+
+        JMenu menu = new JMenu("1 - First Menu");
+        menu.setMnemonic('1');
+        menu.getPopupMenu().addPopupMenuListener(frame);
+
+        menubar.add(menu);
+        for(int i = 0; i < 10; i ++) {
+            menuitem = new JMenuItem("1 JMenuItem" + i);
+            menuitem.setMnemonic('0' + i);
+            menu.add(menuitem);
+        }
+
+        // second menu
+        menu = new JMenu("2 - Second Menu");
+        menu.getPopupMenu().addPopupMenuListener(frame);
+        menu.setMnemonic('2');
+
+        menubar.add(menu);
+        for (int i = 0; i < 5; i++) {
+            menuitem = new JMenuItem("2 JMenuItem" + i);
+            menuitem.setMnemonic('0' + i);
+            menu.add(menuitem);
+        }
+
+        JMenu submenu = new JMenu("Sub Menu");
+        submenu.setMnemonic('S');
+        submenu.getPopupMenu().addPopupMenuListener(frame);
+        for (int i = 0; i < 5; i++) {
+            menuitem = new JMenuItem("S JMenuItem" + i);
+            menuitem.setMnemonic('0' + i);
+            submenu.add(menuitem);
+        }
+        menu.add(new JSeparator());
+        menu.add(submenu);
+
+        return menubar;
+    }
+
+    // PopupMenuListener methods.
+
+    public void popupMenuWillBecomeVisible(PopupMenuEvent e)  {
+        Object source = e.getSource();
+        setMessage("popupmenu visible: " + source.getClass().getName());
```

Comment:
Here and at other locations
```suggestion
        PassFailJFrame.log("popupmenu visible: " + source.getClass().getName());
```